src/firmware/types.rs

```diff
@@ -202,3 +208,522 @@ pub struct SnpPlatformStatus {
+/// | 20     | SINGLE_SOCKET | 0: Guest can be activated on multiple sockets.<br>1: Guest can only be activated on one socket.           |
+///
+#[derive(Default, Clone, Copy, Debug, Serialize, Deserialize)]
+#[repr(C)]
+pub struct SnpGuestPolicy {
+    info: u64,
+}
+
+impl SnpGuestPolicy {
+    /// The minimum ABI minor version required for this guest to run
+    pub fn abi_minor(&self) -> u8 {
+        let mask: u64 = (1 << 8) - 1;
+        (self.info & mask) as u8
+    }
+    /// The minimum ABI major version required for this guest to run
+    pub fn abi_major(&self) -> u8 {
+        let mask: u64 = ((1 << 8) - 1) << 8;
+        ((self.info & mask) >> 8) as u8
+    }
+
+    /// Bit 16 of Guest Policy's info
+    pub fn smt(&self) -> bool {
+        bit_val!(self.info, 16, u64) != 0
+    }
+
+    /// Bit 18 of Guest Policy's info
+    pub fn migrate_ma(&self) -> bool {
+        bit_val!(self.info, 18, u64) != 0
+    }
+
+    /// Bit 19 of Guest Policy's info
+    pub fn debug(&self) -> bool {
+        bit_val!(self.info, 19, u64) != 0
+    }
+
+    /// Bit 20 of Guest Policy's info
+    pub fn single_socket(&self) -> bool {
+        bit_val!(self.info, 20, u64) != 0
+    }
+}
+
+/// The TCB_VERSION is a structure containing the security version numbers of each component in
+/// the trusted computing base (TCB) of the SNP firmware. A TCB_VERSION is associated with each
+/// image of firmware.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+
+pub struct SnpTcbVersion {
+    /// Current bootloader version. SVN of PSP Bootloader.
+    pub boot_loader: u8,
+    /// Current PSP OS version. SVN of PSP Operating System.
+    pub tee: u8,
+    reserved: [u8; 4],
+    /// Version of the SNP firmware. Security Version Number (SVN) of SNP firmware.
+    pub snp: u8,
+    /// Lowest current patch level of all the cores.
+    pub microcode: u8,
+}
+
+impl SnpTcbVersion {
+    pub fn raw(&self) -> String {
+        format!(
+            "{:02}{:02}{:02}{:02}{:02}{:02}{:02}{:02}",
+            self.boot_loader,
+            self.tee,
+            self.reserved[3],
+            self.reserved[2],
+            self.reserved[1],
+            self.reserved[0],
+            self.snp,
+            self.microcode,
+        )
+    }
+}
+
+/// A structure with a bit-field unsigned 64 bit integer:
+/// Bit 0 representing the status of TSME enablement.
+/// Bit 1 representing the status of SMT enablement.
+/// Bits 2-63 are reserved.
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Default, Serialize, Deserialize)]
+pub struct SnpPlatformInfo {
+    info: u64,
+}
+
+impl SnpPlatformInfo {
+    /// Returns the state of TSME
+    pub fn tsme_enabled(&self) -> bool {
+        bit_val!(self.info, 1, u64) != 0
+    }
+
+    /// Returns the state of SMT
+    pub fn smt_enabled(&self) -> bool {
+        bit_val!(self.info, 2, u64) != 0
+    }
+}
```

Comment:
This seems like a use case for `bitflags!`. See it's use [here](https://github.com/virtee/sev/blob/2e54dddffb3ee7cafc805bca203c1136cf41648a/src/launch/snp.rs#L114)